archunit/src/main/java/com/tngtech/archunit/library/freeze/ViolationStoreFactory.java

```diff
@@ -152,21 +152,26 @@ private List<String> replaceCharacter(List<String> violations, String characterT
         }
 
         private String ensureRuleFileName(ArchRule rule) {
+            String ruleDescription = rule.getDescription();
+
             String ruleFileName;
-            if (storedRules.containsKey(rule.getDescription())) {
-                ruleFileName = storedRules.getProperty(rule.getDescription());
-                log.debug("Rule '{}' is already stored in file {}", rule.getDescription(), ruleFileName);
+            if (storedRules.containsKey(ruleDescription)) {
+                ruleFileName = storedRules.getProperty(ruleDescription);
+                log.debug("Rule '{}' is already stored in file {}", ruleDescription, ruleFileName);
             } else {
-                ruleFileName = createNewRuleId(rule).toString();
+                ruleFileName = newRuleFileName(ruleDescription);
+                log.debug("Assigning new ID {} to rule '{}'", ruleFileName, ruleDescription);
+                storedRules.setProperty(ruleDescription, ruleFileName);
             }
             return ruleFileName;
         }
 
-        private UUID createNewRuleId(ArchRule rule) {
-            UUID ruleId = UUID.randomUUID();
-            log.debug("Assigning new ID {} to rule '{}'", ruleId, rule.getDescription());
-            storedRules.setProperty(rule.getDescription(), ruleId.toString());
-            return ruleId;
+        /**
+         * This has <code>protected</code> visibility so that it can be more easily overridden in alternative
```

Comment:
Regardless of whether `com.tngtech.archunit.library.freeze.ViolationStoreFactory.TextFileBasedViolationStore` was designed to be subclassed or not: as it is currently package-private, you can only subclass it within its package anyways.

Reply:
promoted class to top-level